frontend/src/app/chatmenu/BlockUser.tsx

```diff
@@ -1,66 +1,83 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import './BlockUser.css'
-import '../ui/User.css'
 
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { faUnlock, faUserLock } from '@fortawesome/free-solid-svg-icons'
+import { faLock, faLockOpen } from '@fortawesome/free-solid-svg-icons'
 
 import { ChatMenuPageProps } from './ChatMenuPage'
 import { UserData } from '../types/UserData'
+import { Button, UserTile } from '../ui'
+import { fetchFromBackend, useBackendFetchJson } from '../utils'
 
 interface BlockUserProps {
     participants: UserData[]
+    myUid: string
 }
 
-// create user tile to display participants
-const UserTile = ({ user }) => {
-    const { uid, displayName, photoURL } = user
-    const alt = `Profile image for ${displayName}.`
+function useFetchBlockedUsers(myUid: string) {
+    const [res] = useBackendFetchJson({
+        route: `/chat/block/user/${myUid}`,
+        deps: [myUid],
+        isValid: myUid,
+    })
+    const blockedUsers = res?.blocks || []
+    // create a blocked user dictionary
+    return blockedUsers.reduce(
+        (agg, user: { uid: string; blocked: boolean }) => ({
+            ...agg,
+            [user.uid]: user.blocked,
+        }),
+        {}
+    )
+}
 
-    const doBlock = () => {
-        // TODO: Implement blocking the user method
-    }
+async function updateBlock(myUid: string, blockUid: string, shouldBlock: boolean) {
+    await fetchFromBackend({
+        route: `/chat/block/user/${myUid}/${blockUid}/${shouldBlock}`,
+        options: { method: 'POST' },
+    })
+}
+
+// create user tile to display participants
+const CreateUserTile = ({ user, value, submitBlock }) => {
```

Comment:
Optional: Maybe we can rename this component to `BlockUserTile` so that it is a noun instead of a verb?